internal/testutils/xds/e2e/server.go

```diff
@@ -136,7 +137,7 @@ func StartManagementServer(opts ManagementServerOptions) (*ManagementServer, err
 		var err error
 		lis, err = net.Listen("tcp", "localhost:0")
 		if err != nil {
-			return nil, fmt.Errorf("listening on local host and port: %v", err)
+			t.Fatalf("Failed to listen on localhost:0: %v", err)
```

Comment:
Should we call t.Logf() whereever `logger` is called?

Reply:
Done. But in a slightly different way (not storing `testing.T` directly).